### DISCUSSION TITLE
ssh test shows sucess but ssh cloning return permission denied

### DISCUSSION THREAD

**Discussion by keshetjackson** [community/community#73324] [New to GitHub]
ive searched for similar cases but didnt find a solution to help me in this case.
ive been trying to connect my pc to github using ssh and these steps : 

1. open windows shell as admin and cd to shell dir.
2. generate ssh key using `ssh-keygen -t rsa -b 4096 -C "your_email@example.com"`
3. ssh-add 'priavtekeyfile'
4. copy public key and add it as ssh to github thru the settings
5. test a connection getting the message : 'Hi {username}! You've successfully authenticated, but GitHub does not provide shell access.'
6. trying to clone a public repository of mine (same user) using 'git clone {somerepo}

and finally got this message :
'Cloning into 'test'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.'

notes i tried :
  ssh-agent -s = return nothing 
tried this : https://gist.github.com/bsara/5c4d90db3016814a3d2fe38d314f9c23
and this : https://github.com/orgs/community/discussions/33982

**Comment by 0x8** [comment]
Did you ever find a solution to this? I'm having the exact same issue and given I've done nothing complicated to this install, I'm shocked that seemingly nobody else has run into this issue.

As a band-aid solution, I found that installing the Github Desktop application has caused the `git clone` commands to correctly work. The downside is that it will prompt you for the private key every single time (which is a bit annoying). Hopefully someone has a better solution that works without the need for extra applications or hoops.
